src/Factory/Consignment/ConsignmentFactory.php

```diff
@@ -97,16 +100,31 @@ public function fromOrder(
         $this->setOrderData($order, $deliveryOptions);
         $this->createConsignment();
 
-        $collection        = new ConsignmentCollection();
-        $consignment       = $this->initConsignment();
-        $labelAmount       = $this->request['extraOptions']['labelAmount'] ?? 1;
+        $collection   = new ConsignmentCollection();
+        $extraOptions = new ExtraOptions();
+        $consignment  = $this->initConsignment();
+        $labelAmount  = $this->request['extraOptions']['labelAmount'] ?? 1;
+        $extraOptions->setLabelAmount($labelAmount);
         $distributedWeight = $consignment->getTotalWeight() / $labelAmount;
         $consignment->setTotalWeight($distributedWeight);
 
-        for ($i = 0; $i < $labelAmount; $i++) {
-            $collection->addConsignment($consignment);
+        if ($extraOptions->getLabelAmount() > 1) {
+            $countryService = new CountryService();
+            $orderIso       = $countryService->getShippingCountryIso2($order);
```

Comment:
Je hebt de $orderIso niet aangepast, terwijl je toch wel een nieuwe commit moest maken. Zou je een betere naam kunnen gebruiken? Tenzij je echt vindt dat $orderIso een duidelijke naam is, maar als ik dat los lees als variabele naam, heb ik niet een scherp idee wat daarmee bedoeld zou kunnen worden. En wel bij $shippingCountryIso2 bv, of andere opties wellicht.